apis/facebook/api/src/facebook_api.e

```diff
@@ -393,9 +463,8 @@ feature {NONE} -- Implementation
 				create l_raw_file.make_open_read (l_upload_data.file_name.absolute_path.name)
 				if l_raw_file.exists then
 					request.add_header ("Content-Type", l_upload_data.content_type)
-					request.add_header ("Content-Disposition",  "form-data; filename=%"+ l_upload_data.file_name.name%"")
-					request.add_header ("Content-Length", l_raw_file.count.out)
 					request.set_upload_filename (l_upload_data.file_name.absolute_path.name)
+					request.add_form_parameter("source", l_upload_data.file_name.name.as_string_32)
```

Comment:
is `as_string_32` mandatory? doesn't add_form_parameter accepts READABLE_STRING_GENERAL?